src/main/java/net/torocraft/minecoprocessors/util/BookCreator.java

```diff
@@ -17,59 +20,44 @@
 
   private static final String PATH = "/assets/minecoprocessors/books/";
   private static final String PAGE_DELIMITER = "~~~";
+  public static final ItemStack manual = new ItemStack(Items.WRITTEN_BOOK);
 
-  public static ItemStack createBook(String name) {
-    ItemStack book = null;
-    try {
-      book = loadBook(name);
-    } catch (Exception e) {
-      e.printStackTrace();
-    }
-
-    if (book == null) {
-      book = new ItemStack(Items.BOOK);
-    }
-    return book;
-  }
-
-  private static ItemStack loadBook(String name) throws IOException {
-    ItemStack book = new ItemStack(Items.WRITTEN_BOOK);
+  static {
     String line;
     int lineNumber = 1;
     StringBuilder page = newPage();
-    BufferedReader reader = openBookReader(name);
-    while ((line = reader.readLine()) != null) {
-      if (lineNumber == 1) {
-        book.setTagInfo("title", new NBTTagString(line));
-      } else if (lineNumber == 2) {
-        book.setTagInfo("author", new NBTTagString(line));
-      } else if (PAGE_DELIMITER.equals(line)) {
-        writePage(book, page);
-        page = newPage();
-      } else {
-        page.append(line).append("\n");
+    try(BufferedReader reader = openBookReader("manual")) {
+      while ((line = reader.readLine()) != null) {
+        if (lineNumber == 1) {
+          manual.setTagInfo("title", new NBTTagString(line));
+        } else if (lineNumber == 2) {
+          manual.setTagInfo("author", new NBTTagString(line));
+        } else if (PAGE_DELIMITER.equals(line)) {
+          writePage(manual, page);
+          page = newPage();
+        } else {
+          page.append(line).append("\n");
+        }
+        lineNumber++;
       }
-      lineNumber++;
+      writePage(manual, page);
+    } catch(IOException e) {
+      throw new UncheckedIOException(e);
```

Comment:
Neato, I haven't seen this exception wrapper before.

Reply:
Yeah, it's new in Java 8. (For exceptions that aren't IOException, the de-facto standard way of doing this is RuntimeException for things that could possibly happen, and AssertionError for things that can't actually happen (but the Java compiler doesn't know that.))